web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts

```diff
@@ -129,7 +137,17 @@ export function useCreateDatabase() {
       .fetchDatabases(clusterId, request, signal)
       .then(res => {
         if (res.agents.length) {
-          return res.agents[0];
+          // If the db response we got is type AWS,
+          // keep polling until an AWS specific flag
+          // "iamPolicyStatus" returns a non zero enum value.
+          const dbServer = res.agents[0];
+          if (!isAws || !dbServer.aws || !dbServer.aws.iamPolicyStatus) {
```

Comment:
Is there any case where we have `!isAws === true` but also that `dbServer.aws` is defined?

Reply:
no, just wanted to be safe before accessing a possible undefined field for whatever reason